internal/events/aggregator.go

```diff
@@ -535,8 +535,7 @@ func (ag *aggregator) processMessage(ctx context.Context, manifest *core.BatchMa
 		np.IncrementNextPin(ctx, ag.namespace)
 	}
 	state.markMessageDispatched(manifest.ID, msg, msgBaseIndex, newState)
-
-	return err
+	return nil
```

Comment:
Is this a smell that we're not checking the `err = ag.checkOnchainConsistency` error above?
Or else, what path was it that dropped an error down to here that we did **not** want to return?

Reply:
The contract of this method is that it should only return retryable errors, which should happen on line 522. No new errors can happen between that line and here.

Rejection sometimes returns a reason in the error field, which is generally just logged but shouldn't trigger retry.

We probably could spend another few days cleaning up all these paths though.

Reply:
I think the specific path that might return reject with err is via ag.definitions.HandleDefinitionBroadcast.

Reply:
I added another commit to log and drop the error along that specific path, which also likely solves the issues I was seeing. But I think it still feels pretty correct to return nil here, at least as the code stands right now. Figuring out how to spell retryable vs. non-retryable errors throughout this whole package has been challenging.